images/runtime/php-fpm/template.base.Dockerfile

```diff
@@ -12,9 +16,22 @@ ENV NGINX_DOCUMENT_ROOT /home/site/wwwroot
 # - https://www.linuxcapable.com/how-to-install-latest-nginx-mainline-or-stable-on-debian-11/
 RUN apt-get update
 RUN apt install curl nano -y
-RUN curl -sSL https://packages.sury.org/nginx/README.txt | bash -x
-RUN apt-get update
-RUN yes '' | apt-get install nginx-core nginx-common nginx nginx-full -y
+# Install dependencies
+RUN apt-get install -y curl gnupg2 ca-certificates lsb-release
+# Add Nginx's GPG key
+RUN curl -fsSL https://nginx.org/keys/nginx_signing.key | apt-key add -
+# Add the Nginx repository
+RUN echo "deb http://nginx.org/packages/debian `lsb_release -cs` nginx" | tee /etc/apt/sources.list.d/nginx.list
+RUN apt update
+RUN set -eux; \
+    if [ "${DEBIAN_FLAVOR}" = "buster" ]; then \
+        apt install -y nginx=${NGINX_BUSTER_VERSION}; \
+    elif [ "${DEBIAN_FLAVOR}" = "bullseye" ]; then \ 
+        apt install -y nginx=${NGINX_BULLSEYE_VERSION}; \
+    else \
+        apt install -y nginx; \
+    fi
```

Comment:
nit: I know the Dockerfile was structured like this previously, but it's generally best practice in Dockefiles to chain multiple commands into a single `RUN` command rather than having them executed one-by-one as individual `RUN` commands. This will minimize the number of layers that need to be cached and the size of the image as a result.

Reply:
thanks @cormacpayne for sharing the knowledge. I will update the code accordingly.

Reply:
I have addressed the changes. Please have a look whenever..